lib/api/core/models/repositories/repository.js

```diff
@@ -1,344 +1,421 @@
+      return Promise.reject(new InternalError('Bad argument: ' + _ids.toString() + ' is not an array.'));
+    }
+    _ids.forEach(element => {
+      if (!_.isObject(element)) {
+        ids.push(element);
       }
-
-      return response.hits
-        .filter(document => document.found)
-        .map(document => {
-          var object = new this.ObjectConstructor();
-
-          return _.assignIn(object, document._source, {_id: document._id});
-        });
-    });
-};
-
-/**
- * Search in database corresponding repository according to a query
- *
- * @param {object} query
- * @param {number} from manage pagination
- * @param {number} size manage pagination
- *
- * @returns {Promise}
- */
-Repository.prototype.search = function repositorySearch (query, from, size) {
-  return this.databaseEngine.search(this.collection, query, from, size)
-    .then(response => {
-      if (!response.hits || response.hits.length === 0) {
-        return {total: 0, hits: []};
+      else {
+        ids.push(element._id);
       }
-
-      return {
-        total: response.total,
-        hits: response.hits.map((document) => {
-          return _.assignIn({}, document._source, {_id: document._id});
-        })
-      };
     });
-};
 
-/**
- * Loads an object from Cache. Returns a promise that resolves either to the
- * retrieved object of null in case it is not found.
- *
- * The opts object currently accepts one optional parameter: key, which forces
- * the cache key to fetch.
- * In case the key is not provided, it defauls to <collection>/id, i.e.: _kuzzle/users/12
- *
- * @param {string} id The id of the object to get
- * @param {object} opts Optional options.
- * @returns {Promise}
- */
-Repository.prototype.loadFromCache = function repositoryLoadFromCache (id, opts) {
-  var
-    options = opts || {},
-    key = options.key || this.cacheKeyPrefix + id;
+    return this.databaseEngine.mget(collection, ids)
+      .then(response => {
+        if (!response.hits || response.hits.length === 0) {
+          return Promise.resolve([]);
+        }
 
-  return this.cacheEngine.get(key)
-    .then(response => {
-      var object;
+        return response.hits
+          .filter(document => document.found)
+          .map(document => {
+            const object = new this.ObjectConstructor();
 
-      if (response === null) {
-        return null;
-      }
+            return _.assignIn(object, document._source, {_id: document._id});
+          });
+      });
+  }
 
-      object = new this.ObjectConstructor();
+  /**
+   * Search in database corresponding repository according to a query
+   *
+   * @param {object} query
+   * @param {number} from - manage pagination
+   * @param {number} size - manage pagination
+   * @param {string} [collection]
+   * @returns {Promise}
+   */
+  search (query, from, size, collection = this.collection) {
+    if (collection === null) {
+      return Promise.reject(new InternalError('The collection must be specified.'));
```

Comment:
It can not happen due to a bad request, it might come from an bad plugin implementation, will use this one instead.